### ISSUE TITLE
Convert processor cannot be used in autodiscover hints

### ISSUE THREAD

**Issue by ChrsMark** [elastic/beats#17074]
- Version: 7.6, 7.5
- Steps to Reproduce: 

Hints based autodiscover fails to load `convert` configuration if passed with annotation like this:
`co.elastic.logs/processors.0.convert: '{"fields": [{"from": "error", "to": "error.message"}]}'`.

It fails with:
```
ERROR   [autodiscover]  cfgfile/list.go:96      Error creating runner from config: can not convert 'string' into 'object' accessing 'processors.0.convert
```

It is not able to reproduce it with `master`.

cc: @david-kow

**Comment by blakerouse** [comment]
Looks to be a duplicate of #16343.

**Comment by ChrsMark** [comment]
Oh nice, shall we backport #16343 then to `7.6` too? 

cc: @exekias @andresrc

**Comment by ChrsMark** [comment]
@david-kow I don't think we will have any more 7.6.x releases, so it is expected to be fixed with 7.7.

cc: @exekias @andresrc

**Comment by david-kow** [comment]
@ChrsMark Got it, thanks for the update.
